File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        result = []
        for i, num in enumerate(nums):
            if num == 0:
                result.append(num)
            else:
                result.append(nums[(i + (num // abs(num)) * (abs(num) % len(nums)) + len(nums)) % len(nums)])
        return result

File: 3379-transformed-array/3379-transformed-array.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        vector<int> result;
        int n = nums.size();
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] == 0)
            {
                result.push_back(nums[i]);
            }
            else
            {
                result.push_back(nums[(i + (nums[i] / abs(nums[i])) * (abs(nums[i]) % n) + n) % n]);
            }
        }
        return result;
    }
};